sympy/core/equation.py

```diff
@@ -0,0 +1,508 @@
+    >>> q/t
+    c = 1/c
+    >>> t**q
+    a**(a*c) = (b/c)**(b/c**2)
+
+    Utility operations
+    >>> t.reversed
+    b/c = a
+    >>> t.swap
+    b/c = a
+    >>> t.lhs
+    a
+    >>> t.rhs
+    b/c
+    >>> t.as_Boolean()
+    Eq(a, b/c)
+
+    Differentiation
+    Differentiation is applied to both sides if the wrt variable appears on
+    both sides.
+    >>> q=Eqn(a*c, b/c**2)
+    >>> q
+    a*c = b/c**2
+    >>> diff(q,b)
+    Derivative(a*c, b) = c**(-2)
+    >>> diff(q,c)
+    a = -2*b/c**3
+    >>> diff(log(q),b)
+    Derivative(log(a*c), b) = 1/b
+    >>> diff(q,c,2)
+    Derivative(a, c) = 6*b/c**4
+
+    If you specify multiple differentiation all at once the assumption
+    is order of differentiation matters and the lhs will not be
+    evaluated.
+    >>> diff(q,c,b)
+    Derivative(a*c, b, c) = -2/c**3
+
+    To overcome this specify the order of operations.
+    >>> diff(diff(q,c),b)
+    Derivative(a, b) = -2/c**3
+
+    But the reverse order returns an unevaulated lhs (a may depend on b).
+    >>> diff(diff(q,b),c)
+    Derivative(a*c, b, c) = -2/c**3
+
+    Integration can only be performed on one side at a time.
+    >>> q=Eqn(a*c,b/c)
+    >>> integrate(q,b,side='rhs')
+    b**2/(2*c)
+    >>> integrate(q,b,side='lhs')
+    a*b*c
+
+    Make a pretty statement of integration from an equation
+    >>> Eqn(Integral(q.lhs,b),integrate(q,b,side='rhs'))
+    Integral(a*c, b) = b**2/(2*c)
+
+    SymPy's solvers do not understand these equations. They expect an
+    expression that the solver assumes = 0. Thus to use the solver the
+    equation must be rearranged so that all non-zero symbols are on one side.
+    Then just the non-zero symbolic side is passed to ``solve()``.
+    >>> t2 = t-t.rhs
+    >>> t2
+    a - b/c = 0
+    >>> solve(t2.lhs,c)
+    [b/a]
+    """
+
+    def __new__(cls, lhs, rhs, **kwargs):
+        check = kwargs.pop('check', True)
+        lhs = _sympify(lhs)
+        rhs = _sympify(rhs)
+        if not isinstance(lhs,Expr) or not isinstance(rhs,Expr):
+            raise TypeError('lhs and rhs must be valid sympy expressions.')
+        if check and lhs.is_number and rhs.is_number:
+            tst=(lhs-rhs)
+            if tst.is_number:
+                print('Lhs & rhs are numbers. Checking statement validity.'
+                      'Please be patient. This can be slow for complex '
+                      'expressions...')
+                tst=tst.simplify().evalf()
+            if tst != 0 and tst.is_number:
+                from warnings import warn
+                warnstr = '\nDid you really mean to define unequal '
+                warnstr += 'numbers '
+                warnstr += str(lhs) + ' and ' + str(rhs)
+                warnstr += ' as equal?\n'
+                warnstr += 'To suppress this warning include `check=False`'
+                warnstr += ' in the equation definition: '
+                warnstr += '`Eqn(lhs,rhs, check=False)`.'
+                warn(warnstr)
```

Comment:
Here `check` defaults to `True`. This is an absolute dealbreaker as far as I'm concerned: we can not call simplify by default.

We also shouldn't be printing things and giving warnings like this. Remember that sympy is a library as well as a tool for end users. There should be no print statements in a library because downstream applications/libraries that depend on sympy will definitely not want that.

Reply:
Will pull out until I have time to figure out the introspection to make sure it is only used for interactive input.